src/common/os/posix/os_utils.cpp

```diff
@@ -393,4 +395,147 @@ void CtrlCHandler::handler(void*)
 	terminated = true;
 }
 
+namespace {
+
+	class Converter
+	{
+	public:
+		virtual char* convert(const char* fromBuf, size_t fromLen) = 0;
+		virtual void destroy(char* &outBuf) = 0;
+	};
+
+	// No real conversion, system locale is UTF-8 or iconv is not available
+	class NullConverter : public Converter
+	{
+		char* convert(const char* fromBuf, size_t fromLen) override
+		{
+			return const_cast<char*>(fromBuf);
+		}
+		void destroy(char* &outBuf)
+		{
+			outBuf = NULL;
+		}
+	};
+
+#ifdef HAVE_ICONV_H
+	class IConverter : public Converter
+	{
+	public:
+
+		IConverter(const char* sysLocale) : ic(NULL)
```

Comment:
> This is just a copy-paste from isc_file.cpp::IConv class.

class IConv is a helper for calling function iconv(): http://linux.die.net/man/3/iconv
It has nothing to do with interfaces but exactly in this form I treat it's name as good.

What about copy-paste - this is what I do not understand. Why have 2 same (may be almost same) classes with different names? You should have single class performing all actions required from iconv() and use them in all c++ modules where you need. I'm strongly against copyNpaste techology!